src/components/Button/Button.tsx

```diff
@@ -13,26 +13,27 @@ export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
   iconLeft?: IconName;
   iconRight?: IconName;
   align?: Alignment;
-  width?: string;
-  height?: string;
+  fillWidth?: boolean;
+  loading?: boolean;
 }
 
 export const Button = ({
   type = "primary",
   iconLeft,
   iconRight,
-  label,
   align = "center",
   children,
-  width,
-  height,
+  fillWidth,
+  label,
+  loading = false,
+  disabled,
   ...delegated
 }: ButtonProps) => (
   <StyledButton
     $styleType={type}
     $align={align}
-    $width={width}
-    $height={height}
+    $fillWidth={fillWidth}
+    disabled={disabled || loading}
```

Comment:
@gjones Should the loading have disabled style?
I think the loading should show the default style but have only cursor: not-allowed

Reply:
see comment here https://github.com/ClickHouse/control-plane/issues/4669#issuecomment-1694375489